src/main/java/org/jabref/model/groups/TexGroup.java

```diff
@@ -119,6 +120,7 @@ public Path getFilePath() {
     public void fileUpdated() {
         // Reset previous parse result
         keysUsedInAux = null;
+        DefaultTaskExecutor.runInJavaFXThread(() -> metaData.groupsBinding().invalidate());
```

Comment:
In the best case, the invalidation would bubble up automatically from the group along the whole tree. So somehing along the following lines:
- Make each `Group` an observable value that is invalidated when it's properties change. Could be done by making AbstractGroup derive from `PreboundBinding`, change `keysUsedInAux` to an observable set, and invoke `super(keysUsedInAux)` in the constructor
- Invalidate the grouptreenode as soon as the group is invalidated. Should be possible by letting TreeNode derive from PreboundBinding, and then use an extractor here https://github.com/JabRef/jabref/blob/master/src/main/java/org/jabref/model/TreeNode.java#L59 to invalidate the list, when one of the nodes changes.
- Then the root should be invalidated whenever one of the groups reports that is invalidate/changed.

However, that is more work, so I'm fine with this top-down-approach for now as well.

Reply:
I think I'll skip making it observable. It feels like a rabbit-hole that I don't believe I have the JavaFX know-how to dig myself out of (yet). Making `GroupTreeNode#getMatchingGroups` observable after extending with `PreboundBinding` would be another low hanging fruit.

The remaining question would be if from a MVVC point-of-view, does it make more sense to implement the `fileUpdate` method https://github.com/JabRef/jabref/blob/afdb1942c5ec2126e69c92c7f2660459a7bdbeae/src/main/java/org/jabref/model/groups/TexGroup.java#L119
in the `GroupNodeViewModel`, i.e., "initialize" it similar to the `AutomaticPersonsGroup`
https://github.com/JabRef/jabref/blob/afdb1942c5ec2126e69c92c7f2660459a7bdbeae/src/main/java/org/jabref/gui/groups/GroupNodeViewModel.java#L73

Reply:
No, it's fine that the texgroup invalidates the root - that's actually good design, as both of these life in the model. You don't want to take care of this through the UI classes (ViewModel).

Can you please remove the `DefaultTaskExecutor.runInJavaFXThread` part. The BindingsHelper has a `forUI` helper method that allows you consume change events on the JavaFX thread.

Reply:
Just so that I understand you.
The `BindingsHelper.forUI` accepts an `ObservableList` and wraps it in a `UiThreadList` -> I should create an `UiThreadOptionalBinding` wrapper with analogous use to `UiThreadList`/`UiThreadObservableList` and wrap `groupsRootBinding` in it?

I am guessing that means it is ok that any test case that use `setGroup` will get an `Exception in thread "Test worker" java.lang.IllegalStateException: Toolkit not initialized`? (at least that is what happens in `ConvertMarkingToGroupsTest#performMigrationForSingleEntry` on my initial attempt)
I mean, any change in the groups _should_ trigger an UI update.

Reply:
Sure! I'll create an `UiOptionalBinding` wrapper and add a `BindingsHelper#forUI` method for it.

Reply:
I remember we talked once in a devcall about abandoning the google event bus and using observables or properties instead, but I think this is stuff for some far future changes...

Reply:
I might return and look more at it for groups at a later point. I just feel that my JavaFX knowledge is not up to task for it yet. I need to experiment with it more, on something less impactful 😝